superbench/benchmarks/micro_benchmarks/cublas_function/cublas_function_helper.h

```diff
@@ -175,13 +175,25 @@ void from_json(const json &j, CublasFunction &fn) {
         fn.set_batch_count(1);
     }
     try {
-        auto datatype = j.at("datatype").get<std::string>();
-        fn.set_datatype(datatype);
-        auto use_tensor_core = j.at("use_tensor_core").get<bool>();
-        fn.set_use_tensor_core(use_tensor_core);
+        if (str.find("datatype") == std::string::npos) {
+            fn.set_datatype("unknown");
+        } else {
+            auto datatype = j.at("datatype").get<std::string>();
+            fn.set_datatype(datatype);
+        }
+
+    } catch (std::exception &e) {
+        throw std::runtime_error("datatype is required for cublasGemmEx and cublasGemmStridedBatchedEx");
+    }
+    try {
+        if (str.find("use_tensor_core") == std::string::npos) {
```

Comment:
I assume this catches the case where the required value is missing?

I think that, if it is required, it should fail if it is missing.

Reply:
some functions like pure sgemm do not need this parameter, so this parameter is optional not required in some cases